Checkout/Observer/CheckoutObserver.php

```diff
@@ -88,28 +91,18 @@ public function beforeSaveOrder(Varien_Event_Observer $event)
         if (!$boldCheckoutData || $paymentMethod !== Bold_Checkout_Service_PaymentMethod::CODE) {
             return;
         }
-        $taxes = Bold_Checkout_StorefrontClient::call('POST', 'taxes');
-        if (isset($taxes->errors)) {
-            $this->throwException();
-        }
-        if ($order->getDiscountAmount()) {
-            $discounts = Bold_Checkout_StorefrontClient::call(
-                'POST',
-                'discounts',
-                ['code' => 'Discount']
-            );
-            if ($discounts->errors) {
-                $this->throwException();
-            }
-        }
-        $processOrder = Bold_Checkout_StorefrontClient::call('POST', 'process_order');
-        if (isset($processOrder->errors)) {
+        $refreshResult = Bold_Checkout_StorefrontClient::call('GET', 'refresh');
+        $taxesResul = Bold_Checkout_StorefrontClient::call('POST', 'taxes');
```

Comment:
We shouldn't have to call taxes since this will do the exact same thing as the previous /refresh request.

Reply:
If not call 'taxes', 'process_order' will throw error - {
    "errors": [
        {
            "message": "There was an error calculating the taxes",
            "type": "order",
            "field": "tax",
            "severity": "validation",
            "sub_type": "tax"
        }
    ]
}

Reply:
Oh right! Yes you're right.

Reply:
Moved "taxes" and "refresh" call to storefront.